TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/SupportTasks/NpqTrnRequests/Resolve/CheckAnswersTests.cs

```diff
@@ -0,0 +1,594 @@
+
+        await WithDbContext(async dbContext =>
+        {
+            var updatedPersonRecord = await dbContext.Persons
+                .SingleAsync(p => p.PersonId == matchedPerson.PersonId);
+            Assert.Equal(attributeSourcedFromRequestData.Attribute is PersonMatchedAttribute.DateOfBirth ? supportTask.TrnRequestMetadata!.DateOfBirth : matchedPerson.DateOfBirth, updatedPersonRecord.DateOfBirth);
+            Assert.Equal(attributeSourcedFromRequestData.Attribute is PersonMatchedAttribute.EmailAddress ? supportTask.TrnRequestMetadata!.EmailAddress : matchedPerson.Email, updatedPersonRecord.EmailAddress);
+            Assert.Equal(attributeSourcedFromRequestData.Attribute is PersonMatchedAttribute.NationalInsuranceNumber ? supportTask.TrnRequestMetadata!.NationalInsuranceNumber : matchedPerson.NationalInsuranceNumber, updatedPersonRecord.NationalInsuranceNumber);
+        });
+    }
+
+    [Theory]
+    [MemberData(nameof(PersonAttributeInfoData))]
+    public async Task Post_UpdatingExistingRecord_UpdatesSupportTask(PersonAttributeInfo attributeSourcedFromRequestData)
+    {
+        // Arrange
+        var applicationUser = await TestData.CreateApplicationUserAsync();
+
+        var (supportTask, matchedPerson) = await CreateSupportTaskWithSingleDifferenceToMatch(
+            applicationUser.UserId,
+            attributeSourcedFromRequestData.Attribute);
+        var requestData = supportTask.TrnRequestMetadata!;
+
+        var state = new ResolveNpqTrnRequestState()
+        {
+            PersonId = matchedPerson.PersonId,
+            PersonAttributeSourcesSet = true
+        };
+        SetPersonAttributeSourceStateToTrnRequest(state, attributeSourcedFromRequestData.Attribute);
+
+        var journeyInstance = await CreateJourneyInstance(supportTask.SupportTaskReference, state);
+
+        var request = new HttpRequestMessage(
+            HttpMethod.Post,
+            $"/support-tasks/npq-trn-requests/{supportTask.SupportTaskReference}/check-answers?{journeyInstance.GetUniqueIdQueryParameter()}");
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        Assert.Equal(StatusCodes.Status302Found, (int)response.StatusCode);
+
+        // support task is updated
+        await WithDbContext(async dbContext =>
+        {
+            var updatedSupportTask = await dbContext
+                .SupportTasks
+                .Include(st => st.TrnRequestMetadata)
+                .SingleAsync(t => t.SupportTaskReference == supportTask.SupportTaskReference);
+            Assert.Equal(SupportTaskStatus.Closed, updatedSupportTask.Status);
+            Assert.Equal(Clock.UtcNow, updatedSupportTask.UpdatedOn);
+            Assert.Equal(matchedPerson.PersonId, updatedSupportTask.TrnRequestMetadata!.ResolvedPersonId);
+            //Assert.NotNull(updatedSupportTask.TrnRequestMetadata.TrnToken); // CML TODO I don't think I need this TRN token?
```

Comment:
I didn't understand what this token was for - something to do with emails? I concluded I didn't need it - or have I missed something?